sklearn/inspection/_display_estimator.py

```diff
@@ -0,0 +1,274 @@
+  flex-direction: column;
+  align-items: center;
+  float: left;
+  background: white;
+}
+.sk-parallel {
+  display: flex;
+  align-items: stretch;
+  justify-content: center;
+}
+.sk-parallel-item {
+  display: flex;
+  flex-direction: column;
+  position: relative;
+  background: white;
+}
+.sk-parallel-item:first-child::after {
+  align-self: flex-end;
+  width: 50%;
+}
+.sk-parallel-item:last-child::after {
+  align-self: flex-start;
+  width: 50%;
+}
+.sk-parallel-item:only-child::after {
+  width: 0;
+}
+.sk-dashed-wrapped {
+  border: 1px dashed gray;
+  padding: 0 0.25em 0.25em 0.25em;
+}
+.sk-label {
+  font-family: monospace;
+  font-weight: bold;
+  background: white;
+  display: inline-block;
+  margin: 0 0.5em;
+}
+.sk-label-container {
+  text-align: center;
+}
+.sk-serial-item {
+  margin-bottom: 0.25em;
+  background: white;
+}
+.sk-container {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  position: relative;
+  float: left;
+}
+[sk-data-tooltip] {
+  position: relative;
+  cursor: pointer;
+}
+[sk-data-tooltip]:before {
+  visibility: hidden;
+  opacity: 0;
+  font-weight: 400;
+  position: absolute;
+  top: 0;
+  left: 0;
+  padding: 0.5em;
+  overflow: hidden;
+  background-color: #f0f8ff;
+  border: 1px solid gray;
+  white-space: pre;
+  content: attr(sk-data-tooltip);
+  text-align: left;
+}
+[sk-data-tooltip]:hover:before {
+  visibility: visible;
+  opacity: 1;
+  z-index: 2;
+}
+.sk-top-container {
+  display: flex;
+  color: black;
+  padding-bottom: 2em;
+}
+"""
+
+
+def display_estimator(estimator, print_changed_only=True):
+    """Build a HTML representation of an estimator
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator to visualize.
+
+    print_changed_only : bool, optional (default=True)
+        If True, only the parameters that were set to non-default
+        values will be printed when printing an estimator.
+
+    Returns
+    -------
+    html: str or iPython HTML object
+        HTML representation of estimator. When called in jupyter notebook or
+        lab, a iPython HTML object is returned.
+    """
+
+    with config_context(print_changed_only=print_changed_only), \
+            closing(StringIO()) as out:
+
+        if not isinstance(estimator, Pipeline):
+            estimator = Pipeline([(estimator.__class__.__name__, estimator)])
+
+        out.write('<html><head><style>')
+        out.write(_STYLE.replace('\n', ''))
+        out.write('</style></head><body>')
+
+        out.write('<div class="sk-top-container"><div class="sk-container">')
+        _write_estimator_html(out, estimator, '')
+        out.write('</div></div>')  # sk-top-container # sk-container
+        out.write('</body></html>')
+
+        html_output = out.getvalue()
+        # wrap in iPython HTML if in a notebook context
+        try:
+            cls_name = get_ipython().__class__.__name__
+            if cls_name != 'ZMQInteractiveShell':
```

Comment:
I'm wondering if this logic is too magical. Let's say someone built a library which constructed something using the display_estimator output. They may have designed it to expect a string, or am HTML object depending on where they developed it. Use the library in a different context and you get a different result. 

To what extent does returning an object with `_repr_html_` and `__str__` implemented suit our needs? Anyone who wanted to treat it like a string would need to use str() but at least that behaviour would be consistent.